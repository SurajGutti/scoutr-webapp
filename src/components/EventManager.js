import React from 'react';

export const EventsManager =({
    events: {homeTimeOut, awayTimeOut, homeEmptyGoal, awayEmptyGoal,
                toggleHomeTimeout, toggleAwayTimeout,
                toggleHomeEmptyGoal, toggleAwayEmptyGoal}}) => {

    return (
        <div className="ui segment">
            <div className="ui two column very relaxed grid">
                <div className="column">
                    <div className="ui card" style={{ left: '25%', position: 'relative', width:'inherit'}}>
                        <div className="content">
                            <div className="header">Timeout</div>
                            <div className="meta">Home</div>
                                <input id={'homeTimeOut'} type="checkbox" checked={Boolean(homeTimeOut)}
                                       onChange={(event) => toggleHomeTimeout(event.target.checked)}/>
                            <div className="meta">Away</div>
                            <input id={'awayTimeOut'} type="checkbox" checked={Boolean(awayTimeOut)}
                                   onChange={(event) => toggleAwayTimeout(event.target.checked)}/>
                        </div>
                    </div>
                </div>
                <div className="column">
                    <div className="ui card" style={{ left: '25%', position: 'relative', width:'inherit'}}>
                        <div className="content">
                            <div className="header">Empty Goal</div>
                            <div className="meta">Home</div>
                            <input id={'homeEmptyGoal'} type="checkbox" checked={Boolean(homeEmptyGoal)}
                                   onChange={(event) => toggleHomeEmptyGoal(event.target.checked)}/>
                            <div className="meta">Away</div>
                            <input id={'awayEmptyGoal'} type="checkbox" checked={Boolean(awayEmptyGoal)}
                                   onChange={(event) => toggleAwayEmptyGoal(event.target.checked)}/>
                        </div>
                    </div>
                </div>
            </div>
            <div className="ui vertical divider">EVENTS</div>
        </div>
    )
}

// export default Stopwatch;