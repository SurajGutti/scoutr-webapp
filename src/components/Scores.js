import React, {useEffect, useState} from 'react';

export const Scores =({ scores: { homeScore, awayScore, setHomeScore, setAwayScore } }) => {
    return (
        <div style={{textAlign: `center`, marginTop: `100px`}}>
            <div className={"ui segment"}>
                <div className={"ui two column very relaxed grid"}>
                    <div className={"column"}>
                        <div className={"ui labeled input"}>
                            <div className={"ui blue horizontal label"}>Home</div>
                            <input id={'hScore'}
                                   type={'number'}
                                   min={'00'}
                                   max={'99'}
                                   style={{textAlign: 'center'}}
                                   value={homeScore}
                                   onChange={(event) => setHomeScore(+event.target.value)}/>
                        </div>
                    </div>
                    <div className={"column"}>
                        <div className={"ui labeled input"}>
                            <input id={'aScore'}
                                   type={'number'}
                                   min={'00'}
                                   max={'99'}
                                   style={{textAlign: 'center'}}
                                   value={awayScore}
                                   onChange={(event) => setAwayScore(+event.target.value)}/>
                            <div className="ui red horizontal label">Away</div>
                        </div>
                    </div>
                </div>
                <div className={"ui vertical divider"}>
                    VS
                </div>
            </div>
        </div>
    )
}

// export default Stopwatch;