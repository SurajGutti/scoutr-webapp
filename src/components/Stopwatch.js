import React, {useEffect, useState} from 'react';

export const Stopwatch = ({ time: { min, sec, mls, setMin, setSec, setMls, stop, setStop } }) => {
    const [resume, setResume] = useState(false);

    const onStart = () => {
        setStop(false);
    }

    const onResume = () => {
        // console.log(`SetStop: ${stop}`)
        setStop(false);
        setResume(true);
        setTimeout(function () {
            setResume(false);
        }, 500)
    }

    const onStop = () => {
        // console.log(`SetStop: ${stop}`)
        setStop(true);
    }

    const onReset = () => {
        setMin(0);
        setSec(0);
        setMls(0);
        setStop(true);
    }

    const set30 = () => {
        setMin(30);
        setSec(0);
        setMls(0);
    }

    const set60 = () => {
        setMin(60);
        setSec(0);
        setMls(0);
    }

    const set75 = () => {
        setMin(75);
        setSec(0);
        setMls(0);
    }

    useEffect(() => {
        let interval = null;
        if (!stop && (min < 90)){
            if (!resume)
                {
                    if ((min === 30 || min === 60 || min === 75 || min === 90) && sec === 0 && mls === 0) {
                        // console.log(`STOPPED!`);
                        onStop();
                    }
                }
            interval = setInterval(() => {
                if(sec > 59){
                    setMin(min+1);
                    setSec(0);
                    clearInterval(interval);
                }
                if(mls > 99){
                    // console.log(`Seconds:${sec}`);
                    setSec(sec+1);
                    setMls(0);
                    clearInterval(interval);
                }
                if(mls <= 99){
                    setMls(mls+1);
                }
            }, 10)
        }
        else{
            clearInterval(interval);
        }
        return () => {
            clearInterval(interval);
        }
    })

    // onUpdate({min,sec,mls})

    return(
        <div style={{textAlign: "center", marginTop: "50px", marginBottom: `50px`}}>
            <div>
                <h1>
                    {stop && (
                        (<div className={"ui transparent input"}>
                        <input id={'minutes'}
                               type={'number'}
                               min={'00'}
                               max={'89'}
                               style={{textAlign: 'center'}}
                               value={("0"+min).slice(-2)}
                               onChange={(event) => setMin(+event.target.value)}/> :
                        <input id={'seconds'}
                               type={'number'}
                               min={'00'}
                               max={'59'}
                               style={{textAlign: 'center'}}
                               value={("0"+sec).slice(-2)}
                               onChange={(event) => setSec(+event.target.value)}/> :
                        <input id={'millisecs'}
                               type={'number'}
                               min={'00'}
                               max={'99'}
                               style={{textAlign: 'center'}}
                               value={("0"+mls).slice(-2)}
                               onChange={(event) => setMls(+event.target.value)}/></div>))}
                    {!stop && (<div><span style={{textAlign: 'center'}}>{("0"+min).slice(-2)}</span> :
                        <span style={{textAlign: 'center'}}>{("0"+sec).slice(-2)}</span> :
                        <span style={{textAlign: 'center'}}>{("0"+mls).slice(-2)}</span></div>)}
                </h1>
                {stop && (min === 0) && (sec === 0) && (mls === 0) && (<div className={"ui icon buttons"}><button
                    className="ui button" onClick={onStart}><i className={"play icon"}></i></button></div>)}
                {stop && ((min !== 0) || (sec !== 0) || (mls !== 0)) && (
                    <div className={"ui icon buttons"}>
                        <button className={"ui button"} onClick={onResume}>
                            <i className="play circle outline icon"></i>
                        </button>
                    </div>
                    )}
                {!stop && (<div className={"ui icon buttons"} >
                    <button className={"ui button"} onClick={onStop}>
                        <i className="pause icon"></i>
                    </button>
                </div>)}
            </div><br/>
            <div className="ui buttons">
                <button className="ui button" onClick={onReset}>00</button>
                <button className="ui button" onClick={set30}>30</button>
                <button className="ui button" onClick={set60}>60</button>
                <button className="ui button" onClick={set75}>75</button>
            </div>
        </div>
    )
}

// export default Stopwatch;