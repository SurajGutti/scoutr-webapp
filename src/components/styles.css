.backdrop{
    background: rgb(255, 255, 255);
    background: linear-gradient(180deg, rgba(244,242,255,1) 0%, rgb(225, 224, 224) 50%, rgb(202, 203, 205) 100%);
    height: 100%;
}

.mainheader{
    font-family: Verdana, Geneva, sans-serif;;
    font-size: 40px;
    letter-spacing: 6px;
    word-spacing: -5px;
    color: rgba(55, 67, 96, 0.59);
    font-weight: normal;
    text-decoration: none;
    font-variant: small-caps;
    text-transform: none;
}

.mainheader .sub{
    font-family: Verdana, Geneva, sans-serif;
    font-size: 25px;
    letter-spacing: 0;
    word-spacing: 2.4px;
}

.homepage {
    position: fixed;
    width: 500px;
    height: 200px;
    top: 50%;
    left: 60%;
    margin-top: -100px;
    margin-left: -250px;
}

.error {
    position: relative;
    padding-bottom: 0;
    padding-top: 1rem;
    font-family: Verdana, Geneva, sans-serif;
    color: darkred;
    font-weight: normal;
}