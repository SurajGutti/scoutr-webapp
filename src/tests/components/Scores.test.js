import { shallow } from "enzyme";
import { Scores } from "../../components/Scores";

test('testing Scores Component', () => {
   const setHomeScore = jest.fn();
   const setAwayScore = jest.fn();
   const scores = {
      homeScore: 60,
      awayScore: 30,
      setHomeScore: setHomeScore,
      setAwayScore: setAwayScore
   };
   const wrapper = shallow(<Scores scores={scores} />);
   expect(wrapper).toMatchSnapshot();
   wrapper.find('#hScore').simulate('change', {
      target: {
         value: scores.homeScore
      }
   });
   wrapper.find('#aScore').simulate('change', {
      target: {
         value: scores.awayScore
      }
   })
});