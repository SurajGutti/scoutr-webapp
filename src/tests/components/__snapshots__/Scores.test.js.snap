// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Scores Component 1`] = `
<div
  style={
    Object {
      "marginTop": "100px",
      "textAlign": "center",
    }
  }
>
  <div
    className="ui segment"
  >
    <div
      className="ui two column very relaxed grid"
    >
      <div
        className="column"
      >
        <div
          className="ui labeled input"
        >
          <div
            className="ui blue horizontal label"
          >
            Home
          </div>
          <input
            id="hScore"
            max="99"
            min="00"
            onChange={[Function]}
            style={
              Object {
                "textAlign": "center",
              }
            }
            type="number"
            value={60}
          />
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui labeled input"
        >
          <input
            id="aScore"
            max="99"
            min="00"
            onChange={[Function]}
            style={
              Object {
                "textAlign": "center",
              }
            }
            type="number"
            value={30}
          />
          <div
            className="ui red horizontal label"
          >
            Away
          </div>
        </div>
      </div>
    </div>
    <div
      className="ui vertical divider"
    >
      VS
    </div>
  </div>
</div>
`;
